input/script_utils.go

```diff
@@ -249,6 +249,109 @@ func SenderHTLCScript(senderHtlcKey, receiverHtlcKey,
 	return builder.Script()
 }
 
+// SenderHTLCScriptAnchors constructs the public key script for an outgoing
+// HTLC output payment for the sender's version of the commitment transaction,
+// for the anchor channel type. It is equal to the SenderHTLCScript except an
+// added 1 block delay in the non-revocation cases.
+//
+// OP_DUP OP_HASH160 <revocation key hash160> OP_EQUAL
+// OP_IF
+//     OP_CHECKSIG
+// OP_ELSE
+//     <recv htlc key>
+//     OP_SWAP OP_SIZE 32 OP_EQUAL
+//     OP_NOTIF
+//         OP_DROP 2 OP_SWAP <sender htlc key> 2 OP_CHECKMULTISIG
+//     OP_ELSE
+//         OP_HASH160 <ripemd160(payment hash)> OP_EQUALVERIFY
+//         OP_CHECKSIG
+//     OP_ENDIF
+//     1 OP_CHECKSEQUENCEVERIFY
+// OP_ENDIF
+func SenderHTLCScriptAnchors(senderHtlcKey, receiverHtlcKey,
```

Comment:
for these scripts we could pass a boolen/chantype in to determine whether to add the OP_CSV, but copied for now to make them totally independent.